src/transformers/api-history.ts

```diff
@@ -0,0 +1,189 @@
+import { Code, Node } from 'mdast';
+import { getJSXImport, isCode, isImport } from '../util/mdx-utils';
+import { ActionTuple, SKIP, visitParents } from 'unist-util-visit-parents';
+import { Parent } from 'unist';
+import { parse as parseYaml } from 'yaml';
+import AdmZip from 'adm-zip';
+
+export type ApiHistory = {
+  added?: { 'pr-url': string }[];
+  deprecated?: { 'pr-url': string; 'breaking-changes-header': string }[];
+  changes?: { 'pr-url': string; description: string }[];
+};
+
+export interface PrReleaseVersions {
+  release: string | null;
+  backports: Array<string>;
+}
+export type PrReleaseVersionsContainer = { [key: number]: PrReleaseVersions };
+
+interface PrReleaseArtifact {
+  data: PrReleaseVersionsContainer;
+  endCursor: string;
+}
+
+export default function attacher() {
+  return transformer;
+}
+
+function matchApiHistoryCodeBlock(node: Node): node is Code {
+  return isCode(node) && node.lang === 'YAML' && node.meta === 'history';
+}
+
+let _allPrReleaseVersions: PrReleaseVersionsContainer = null;
+
+async function getAllPrReleaseVersions(): Promise<PrReleaseVersionsContainer> {
+  if (_allPrReleaseVersions) {
+    return _allPrReleaseVersions;
+  }
+
+  // TODO: Add error handling and logging
+  if (process.env.GITHUB_ACTIONS === 'true' && !process.env.GH_TOKEN) {
+    throw new Error('GH_TOKEN is required when running in GitHub Actions.');
+  }
+
+  // TODO: Remove this
+  if (process.env.LOCAL_DEV === 'true') {
+    const versions: PrReleaseVersionsContainer = {
+      35658: {
+        release: 'v30.0.0-nightly.20231214',
+        backports: ['v29.0.0-alpha.9'],
+      },
+      41391: {
+        release: 'v31.0.0-alpha.1',
+        backports: [] as string[],
+      },
+      42086: {
+        release: 'v32.0.0-nightly.20240531',
+        backports: ['v31.0.0'],
+      },
+    };
+
+    _allPrReleaseVersions = versions;
+    return _allPrReleaseVersions;
+  }
```

Comment:
@dsanders11 Should I make the `LOCAL_DEV` case default for now so it shows up in the Heroku deploys?

Reply:
@piotrpdev, for the immediate future that might be the best short-term solution. I don't know enough about the Heroku preview deployments to know if we can provide it with a GH token, I'll look into it.

Reply:
Made default in [`fd76258` (#594)](https://github.com/electron/website/pull/594/commits/fd7625898ccd26bd48da1022b7208946a54b2d05) for now.